app/controllers/application_controller.rb

```diff
@@ -322,7 +322,9 @@ def saved_redirect_url_or(default)
 
   def generic_exception(exception)
     Foreman::Logging.exception("Action failed", exception)
-    render :template => "common/500", :layout => !request.xhr?, :status => :internal_server_error, :locals => { :exception => exception}
+    ex_message = exception.message
+    exception_code = Foreman::Exception.calculate_error_code(self.class.to_s, ex_message)
```

Comment:
I'm not sure this would give the correct code calculation. 
In case the exception is already a Foreman::Exception, the message will already include the ERF code (so this will generate a different code), while if this is a different kind of exception the calculation will give a different code than you would expect since class would be the controller and not the actual class that threw the exception.